Migrating GCE Bill Analaysis to BigQuery

namrathaurs opened the pull request:
Fixes HEPCloud/decisionengine_modules#458.

namrathaurs committed:
migrating gce billing to bigquery

- removed code that is obsolete since cloud billing data transitioned to
  bigquery
- introduced the use of bigquery client APIs to interact with bigquery
  to aid the bill calculation logic
- removed some of the configuration-related details since they were not
  being used previously (i.e. pre-bigquery)

namrathaurs committed:
minor changes

- cleaned up unused import statements and orderred them as per the PEP
  standards
- fixed indentation of the newly added class method

namrathaurs committed:
migrating to using bigquery for GCE cloud billing

- refactored existing code to remove unused lines of code and import
  statements
- modified logic to make use of BigQuery APIs to compute costs incurred
  from using Google Cloud services

pep8speaks commented:
Hello @namrathaurs! Thanks for updating this PR. We checked the lines you've touched for [PEP 8](https://www.python.org/dev/peps/pep-0008) issues, and found:

* In the file [`src/bill_calculator_hep/GCEBillAnalysis.py`](https://github.com/HEPCloud/bill-calculator-hep/blob/770164c611ef600ae8fc1f52b2925e930fbf5f12/src/bill_calculator_hep/GCEBillAnalysis.py):

> [Line 27:48](https://github.com/HEPCloud/bill-calculator-hep/blob/770164c611ef600ae8fc1f52b2925e930fbf5f12/src/bill_calculator_hep/GCEBillAnalysis.py#L27): [E261](https://duckduckgo.com/?q=pep8%20E261) at least two spaces before inline comment
> [Line 27:49](https://github.com/HEPCloud/bill-calculator-hep/blob/770164c611ef600ae8fc1f52b2925e930fbf5f12/src/bill_calculator_hep/GCEBillAnalysis.py#L27): [E262](https://duckduckgo.com/?q=pep8%20E262) inline comment should start with '# '
> [Line 46:80](https://github.com/HEPCloud/bill-calculator-hep/blob/770164c611ef600ae8fc1f52b2925e930fbf5f12/src/bill_calculator_hep/GCEBillAnalysis.py#L46): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (128 > 79 characters)
> [Line 48:80](https://github.com/HEPCloud/bill-calculator-hep/blob/770164c611ef600ae8fc1f52b2925e930fbf5f12/src/bill_calculator_hep/GCEBillAnalysis.py#L48): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (80 > 79 characters)
> [Line 60:80](https://github.com/HEPCloud/bill-calculator-hep/blob/770164c611ef600ae8fc1f52b2925e930fbf5f12/src/bill_calculator_hep/GCEBillAnalysis.py#L60): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (99 > 79 characters)
> [Line 81:80](https://github.com/HEPCloud/bill-calculator-hep/blob/770164c611ef600ae8fc1f52b2925e930fbf5f12/src/bill_calculator_hep/GCEBillAnalysis.py#L81): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (80 > 79 characters)
> [Line 83:80](https://github.com/HEPCloud/bill-calculator-hep/blob/770164c611ef600ae8fc1f52b2925e930fbf5f12/src/bill_calculator_hep/GCEBillAnalysis.py#L83): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (120 > 79 characters)
> [Line 85:80](https://github.com/HEPCloud/bill-calculator-hep/blob/770164c611ef600ae8fc1f52b2925e930fbf5f12/src/bill_calculator_hep/GCEBillAnalysis.py#L85): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (105 > 79 characters)
> [Line 87:80](https://github.com/HEPCloud/bill-calculator-hep/blob/770164c611ef600ae8fc1f52b2925e930fbf5f12/src/bill_calculator_hep/GCEBillAnalysis.py#L87): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (111 > 79 characters)
> [Line 88:80](https://github.com/HEPCloud/bill-calculator-hep/blob/770164c611ef600ae8fc1f52b2925e930fbf5f12/src/bill_calculator_hep/GCEBillAnalysis.py#L88): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (127 > 79 characters)
> [Line 90:80](https://github.com/HEPCloud/bill-calculator-hep/blob/770164c611ef600ae8fc1f52b2925e930fbf5f12/src/bill_calculator_hep/GCEBillAnalysis.py#L90): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (82 > 79 characters)
> [Line 108:80](https://github.com/HEPCloud/bill-calculator-hep/blob/770164c611ef600ae8fc1f52b2925e930fbf5f12/src/bill_calculator_hep/GCEBillAnalysis.py#L108): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (123 > 79 characters)
> [Line 115:80](https://github.com/HEPCloud/bill-calculator-hep/blob/770164c611ef600ae8fc1f52b2925e930fbf5f12/src/bill_calculator_hep/GCEBillAnalysis.py#L115): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (80 > 79 characters)
> [Line 116:80](https://github.com/HEPCloud/bill-calculator-hep/blob/770164c611ef600ae8fc1f52b2925e930fbf5f12/src/bill_calculator_hep/GCEBillAnalysis.py#L116): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (92 > 79 characters)
> [Line 121:80](https://github.com/HEPCloud/bill-calculator-hep/blob/770164c611ef600ae8fc1f52b2925e930fbf5f12/src/bill_calculator_hep/GCEBillAnalysis.py#L121): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (319 > 79 characters)
> [Line 122:47](https://github.com/HEPCloud/bill-calculator-hep/blob/770164c611ef600ae8fc1f52b2925e930fbf5f12/src/bill_calculator_hep/GCEBillAnalysis.py#L122): [E262](https://duckduckgo.com/?q=pep8%20E262) inline comment should start with '# '
> [Line 123:47](https://github.com/HEPCloud/bill-calculator-hep/blob/770164c611ef600ae8fc1f52b2925e930fbf5f12/src/bill_calculator_hep/GCEBillAnalysis.py#L123): [E262](https://duckduckgo.com/?q=pep8%20E262) inline comment should start with '# '
> [Line 123:80](https://github.com/HEPCloud/bill-calculator-hep/blob/770164c611ef600ae8fc1f52b2925e930fbf5f12/src/bill_calculator_hep/GCEBillAnalysis.py#L123): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (151 > 79 characters)
> [Line 126:80](https://github.com/HEPCloud/bill-calculator-hep/blob/770164c611ef600ae8fc1f52b2925e930fbf5f12/src/bill_calculator_hep/GCEBillAnalysis.py#L126): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (83 > 79 characters)
> [Line 132:80](https://github.com/HEPCloud/bill-calculator-hep/blob/770164c611ef600ae8fc1f52b2925e930fbf5f12/src/bill_calculator_hep/GCEBillAnalysis.py#L132): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (88 > 79 characters)
> [Line 135:49](https://github.com/HEPCloud/bill-calculator-hep/blob/770164c611ef600ae8fc1f52b2925e930fbf5f12/src/bill_calculator_hep/GCEBillAnalysis.py#L135): [E262](https://duckduckgo.com/?q=pep8%20E262) inline comment should start with '# '
> [Line 137:49](https://github.com/HEPCloud/bill-calculator-hep/blob/770164c611ef600ae8fc1f52b2925e930fbf5f12/src/bill_calculator_hep/GCEBillAnalysis.py#L137): [E262](https://duckduckgo.com/?q=pep8%20E262) inline comment should start with '# '
> [Line 143:80](https://github.com/HEPCloud/bill-calculator-hep/blob/770164c611ef600ae8fc1f52b2925e930fbf5f12/src/bill_calculator_hep/GCEBillAnalysis.py#L143): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (535 > 79 characters)
> [Line 175:80](https://github.com/HEPCloud/bill-calculator-hep/blob/770164c611ef600ae8fc1f52b2925e930fbf5f12/src/bill_calculator_hep/GCEBillAnalysis.py#L175): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (95 > 79 characters)
> [Line 177:80](https://github.com/HEPCloud/bill-calculator-hep/blob/770164c611ef600ae8fc1f52b2925e930fbf5f12/src/bill_calculator_hep/GCEBillAnalysis.py#L177): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (156 > 79 characters)
> [Line 178:80](https://github.com/HEPCloud/bill-calculator-hep/blob/770164c611ef600ae8fc1f52b2925e930fbf5f12/src/bill_calculator_hep/GCEBillAnalysis.py#L178): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (80 > 79 characters)
> [Line 185:68](https://github.com/HEPCloud/bill-calculator-hep/blob/770164c611ef600ae8fc1f52b2925e930fbf5f12/src/bill_calculator_hep/GCEBillAnalysis.py#L185): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 185:70](https://github.com/HEPCloud/bill-calculator-hep/blob/770164c611ef600ae8fc1f52b2925e930fbf5f12/src/bill_calculator_hep/GCEBillAnalysis.py#L185): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 190:80](https://github.com/HEPCloud/bill-calculator-hep/blob/770164c611ef600ae8fc1f52b2925e930fbf5f12/src/bill_calculator_hep/GCEBillAnalysis.py#L190): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (102 > 79 characters)
> [Line 195:80](https://github.com/HEPCloud/bill-calculator-hep/blob/770164c611ef600ae8fc1f52b2925e930fbf5f12/src/bill_calculator_hep/GCEBillAnalysis.py#L195): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (102 > 79 characters)
> [Line 200:80](https://github.com/HEPCloud/bill-calculator-hep/blob/770164c611ef600ae8fc1f52b2925e930fbf5f12/src/bill_calculator_hep/GCEBillAnalysis.py#L200): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (88 > 79 characters)



##### Comment last updated at 2024-07-25 19:17:31 UTC

namrathaurs committed:
addressing PEP8 issues

- converted long lines of comments to hard wrapped lines at 80
  characters per line
